app/services/twilio/incoming_message_service.rb

```diff
@@ -106,15 +106,25 @@ def additional_attributes
   end
 
   def attach_files
-    return if params[:MediaUrl0].blank?
+    num_media = params[:NumMedia].to_i
+    return if num_media.zero?
 
-    attachment_file = download_attachment_file
+    num_media.times do |i|
+      attach_single_file(i)
+    end
+  end
+
+  def attach_single_file(index)
+    media_url = params[:"MediaUrl#{index}"]
+    content_type = params[:"MediaContentType#{index}"]
+    return if media_url.blank?
 
+    attachment_file = download_attachment_file(index)
     return if attachment_file.blank?
 
     @message.attachments.new(
       account_id: @message.account_id,
-      file_type: file_type(params[:MediaContentType0]),
+      file_type: file_type(content_type),
```

Comment:
```suggestion
      file_type: file_type(attachment_file.content_type),
```

We can use this directly